samples/vboxwrapper/vbox_common.cpp

```diff
@@ -262,51 +283,206 @@ void VBOX_BASE::dump_hypervisor_logs(bool include_error_logs) {
+		    This might be a temporary problem and so this job will be rescheduled for another time.\n",
+		
+		"   NOTE: BOINC has detected that your computer's processor supports hardware acceleration for\n \
+		    virtual machines but the hypervisor failed to successfully launch with this feature enabled.\n \
+		    This means that the hardware acceleration feature has been disabled in the computer's BIOS.\n \
+		    Please enable this feature in your computer's BIOS.\n \
+		    Intel calls it 'VT-x'\n \
+		    AMD calls it 'AMD-V'\n \
+		    More information can be found here: https://en.wikipedia.org/wiki/X86_virtualization\n \
+		    Error Code: ERR_CPU_VM_EXTENSIONS_DISABLED\n",
+		
+		"   NOTE: VirtualBox has reported an improperly configured virtual machine. It was configured to require\n \
+		    hardware acceleration for virtual machines, but your processor does not support the required feature.\n \
+		    Please report this issue to the project so that it can be addresssed.\n",
+		
+		"   NOTE: VirtualBox hypervisor reports that another hypervisor has locked the hardware acceleration\n \
+		    for virtual machines feature in exclusive mode.\n",
+		
+		"   NOTE: VirtualBox has failed to allocate enough memory to start the configured virtual machine.\n \
+		    This might be a temporary problem and so this job will be rescheduled for another time.\n",
+		
+		"   NOTE: VM failed to enter an online state within the timeout period.\n \
+		    This might be a temporary problem and so this job will be rescheduled for another time.\n",
+	
+		    "VM environment needed to be cleaned up.",
+       
+		    "Forign VM Hypervisor locked hardware acceleration features.",
+
+		    "VM Hypervisor was unable to allocate enough memory to start VM.",
+
+	     	"VM Hypervisor failed to enter an online state in a timely fashion."
+	};
+
+	std::vector<std::string> v(args, args + 14);
+	
+	return v[num];
+}
+
+string VBOX_BASE::read_vm_log(){
+	string line;
+	size_t line_pos;
+	size_t line_end;
+	size_t line_start;
+	string msg;
+	string virtualbox_vm_log;
+	virtualbox_vm_log = vm_master_name + "/Logs/VBox.log";
+	
+
+	std::ifstream  src(virtualbox_vm_log.c_str(), std::ios::binary);
+	    
+	if ((src.is_open()) && (src.seekg(log_pointer))) {
+
+            while (std::getline(src, line)) {
+
+                line_start = line.find("Console:");
+		if (line_start != string::npos) {
+					
+                    line_start += 35;
+                    line_end = line.find("\'", line_start);
+                    msg = line.substr(line_start, line_end - line_start);
+                    sanitize_format(msg);
+                    state = msg.c_str();
+                }
+                line_pos = line.find("Guest Log:");
+                if (line_pos != string::npos) {
+					
+                    msg = line.substr(line_pos, line.size() - line_pos);
+                    sanitize_format(msg);
+                    vboxlog_msg(msg.c_str());
```

Comment:
It's not entirely clear to me if it's required to open the file in binary mode in order to use `tellg` and `seekg` on Windows. Assuming it is, you need to remove `'\r'` from the end of the message. There appears to be `sanitize_output` that does it.